src/lib/Microsoft.Health.Events/Telemetry/Metrics/EventMetrics.cs

```diff
@@ -95,15 +67,19 @@ public static Metric EventsConsumed(EventMetricDefinition eventMetricDefinition,
         /// <param name="dateTime">The datetime of the watermark</param>
         public static Metric EventWatermark(string partitionId, DateTime dateTime)
         {
-            return new Metric(
-                "EventsWatermarkUpdated",
-                new Dictionary<string, object>
-                {
-                    { _nameDimension, "EventsWatermarkUpdated" },
-                    { _timeDimension, dateTime.ToString() },
-                    { _partitionDimension, partitionId },
-                    { _categoryDimension, Category.Latency },
-                });
+            return CreateBaseEventMetric(EventMetricNames.EventsWatermarkUpdated, partitionId, Category.Traffic)
+                .AddDimension(_timeDimension, dateTime.ToString());
+        }
+
+        /// <summary>
+        /// Signals the timestamp corresponding to the last processed event per partition.
+        /// </summary>
+        /// <param name="partitionId">The partition id of the event hub</param>
+        /// <param name="triggerReason">The trigger that caused the events to be flushed and processed </param>
+        public static Metric EventTimestampLastProcessedPerPartition(string partitionId, string triggerReason)
+        {
+            return CreateBaseEventMetric(EventMetricNames.EventTimestampLastProcessedPerPartition, partitionId, Category.Latency)
```

Comment:
I went with category Latency since we can use this to figure out if we are behind in processing. But feels like it also overlaps a little with Traffic since depending on the incoming events, this will update. Open to change if needed.

Reply:
The category makes sense to me.